### ISSUE TITLE
randomly completion during GRPO training

### ISSUE THREAD

**Issue by wufan-tb** [modelscope/ms-swift#6957]
**Describe the bug**
I'm trying train Qwen3VL with GRPO, yet the completion during training are randomly generated:

<img width="1994" height="599" alt="Image" src="https://github.com/user-attachments/assets/c997e170-6a1e-408a-9d2d-8f48fe5db990" />

the prompt seems abnormal neither, it seems missed the image token, I'm not sure.

what is the reason for this, and how I can fix it? any help would be appreciated.

**Your hardware and system info**
system : I used the official docker image: modelscope-registry.cn-hangzhou.cr.aliyuncs.com/modelscope-repo/modelscope:ubuntu22.04-cuda12.8.1-py311-torch2.8.0-vllm0.11.0-modelscope1.31.0-swift3.10.3

GPU: NVIDIA H200

**Additional context**
self defined dataset:
```
{
    "images": [
        "xxx"
    ],
    "messages": [
        {
            "role": "user",
            "content": "Please convert the table in this image into valid OTSL format:"
        }
    ],
    "solution": "xxx"
}
```

**Comment by hjh0119** [comment]
please provide reproduce script and env infos

**Comment by wufan-tb** [comment]
rollout script:

```
CUDA_VISIBLE_DEVICES=4,5 \
nohup swift rollout \
    --model ~/Qwen3-VL-2B-OTSL_SFT \
    --vllm_tensor_parallel_size 2 \
    --vllm_data_parallel_size 1 \
    --vllm_gpu_memory_utilization 0.85 \
    --vllm_max_model_len 16384 \
    > logs/rollout.log 2>&1 &
```

grpo script:

```
NPROC_PER_NODE=4 \
CUDA_VISIBLE_DEVICES=0,1,2,3 \
nohup swift rlhf \
    --rlhf_type grpo \
    --model $base_model \
    --external_plugins plugin/plugin.py \
    --reward_funcs oeds_reward \
    --vllm_mode server \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8000 \
    --train_type full \
    --torch_dtype bfloat16 \
    --attn_impl flash_attn \
    --dataset '/self defined dataset.jsonl#5000' \
    --split_dataset_ratio 0 \
    --max_completion_length 8192 \
    --num_train_epochs $num_epoch \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 4 \
    --learning_rate 1e-5 \
    --save_strategy 'steps' \
    --eval_strategy 'steps' \
    --save_steps 200 \
    --save_total_limit 1 \
    --logging_steps 100 \
    --output_dir $output_dir \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 16 \
    --num_generations 8 \
    --temperature 1.0 \
    --deepspeed zero2 \
    --log_completions true \
    --num_iterations 1 \
    --async_generate false \
    --beta 0.001 \
    --use_hf True \
    > logs/grpo.log 2>&1 &
```

env: I used this docker image:  `modelscope-registry.cn-hangzhou.cr.aliyuncs.com/modelscope-repo/modelscope:ubuntu22.04-cuda12.8.1-py311-torch2.8.0-vllm0.11.0-modelscope1.31.0-swift3.10.3`

**Comment by hjh0119** [comment]
`export IMAGE_MAX_TOKEN_NUM=1536` for both rollout and GRPO.

**Comment by wufan-tb** [comment]
> `export IMAGE_MAX_TOKEN_NUM=1536` for both rollout and GRPO.

I tried this apporch, but it's not work for me.

**Comment by hjh0119** [comment]
Check whether the first training step is normal; if so, try reducing the learning rate.

**Comment by wufan-tb** [comment]
this phenomenon appeared at first training step.

**Comment by hjh0119** [comment]
plz check:
- Whether the model produces normal outputs when directly inferring on the dataset (e.g., prematurely outputting the EOS token).
- Whether the base model (before sft) reproduce

**Comment by wufan-tb** [comment]
> plz check:
> 
> * Whether the model produces normal outputs when directly inferring on the dataset (e.g., prematurely outputting the EOS token).
> * Whether the base model (before sft) reproduce

- I tried the base model (qwen3vl 2b), and changed prompt (OTSL format to HTML format), but this phenomenon still appedred.

<img width="1992" height="864" alt="Image" src="https://github.com/user-attachments/assets/823351d9-9688-4c4c-8db9-945df677d70d" />

- I'm not familiar with how to inferring on dataset, but I tried infer on table images by vllm, it's outputs are normal.

**Comment by hjh0119** [comment]
> I tried the base model (qwen3vl 2b), and changed prompt (OTSL format to HTML format), but this phenomenon still appedred.

Could you please provide some samples of the dataset and the prompt so I can reproduce the issue?
Alternatively, does it occur on any open-source dataset?

**Comment by wufan-tb** [comment]
of course, here are two simple demo sample:

[sample.zip](https://github.com/user-attachments/files/24052682/sample.zip)

thank you for your kindness!

**Comment by hjh0119** [comment]
--use_vllm true

**Comment by wufan-tb** [comment]
> --use_vllm true

I accidentally omitted this parameter!  the problem solved, thanks very much!
